### PR TITLE
INS-4375/evaluate rules

### PR DESCRIPTION
**Pull request by amr-alaa95**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,4 +3,8 @@
 .vscode
 events_emitter/settings.py
 *.sqlite3
-**/__pycache/**
\ No newline at end of file
+**/__pycache/**
+.env
+celerybeat-schedule
+staticfiles/
+**/__pycache/**
```
Comment (mhmdsamir92) [inline .gitignore]: duplicate with line 6

File: events_emitter/celery.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+from django.conf import settings
+
+os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'events_emitter.settings')
+
+app = Celery('events_emitter')
+
+app.config_from_object(settings, namespace='CELERY')
+
+app.conf.task_routes = {
+    'events_emitter.tasks*': {'queue': settings.EVENTS_EMITTER_QUEUE},
```
Comment (mhmdsamir92) [inline events_emitter/celery.py]: Why we don't just use the default celery queue, I think we don't need to specify the queue name unless there are multiple queues needed

File: run.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+set -x
+
+if [ "$1" = "events_emitter" ]; then
+    echo "==================="
+    echo "Running events_emitter"
+    echo "==================="
+    echo
+    python manage.py migrate
+    python manage.py runserver 0.0.0.0:9000
```
Comment (mhmdsamir92) [inline run.sh]: Please use gunicorn

File: events_emitter/tasks.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    proccess_rules_group = chain(group(proccess_rule_list), proccess_rules_res.s())
+    proccess_rules_group.apply_async(queue=settings.EVENTS_EMITTER_QUEUE)
+
+
+@shared_task
+def proccess_rule(rule, *args, **kwargs):
+    logger.info(f"start evaluate rule {rule.get('event_type')}")
+    rule_res = False
+    try:
+        time_series_type = FactoryEvent().create_event_class(settings.TIME_SERIES_TYPE)
```
Comment (AlaaNour94) [inline events_emitter/tasks.py]: can we change this variable name into something more meaningful eg "time_series_datastore"

File: events_emitter/example-settings.py [hunk 5]
```diff
@@ -11,15 +11,17 @@
...
+import environ
 
 getenv = os.getenv
 ENV = getenv('ENVIRONMENT', 'development')
 
 # Build paths inside the project like this: os.path.join(BASE_DIR, ...)
 BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
-
-
+ROOT_DIR = environ.Path(__file__) - 3
```
Comment (AlSayedGamal) [inline events_emitter/example-settings.py]: what's -3 ?

File: events_emitter/example-settings.py [hunk 6]
```diff
@@ -29,7 +31,7 @@
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = True
 
-ALLOWED_HOSTS = []
+ALLOWED_HOSTS = ['*']
```
Comment (AlSayedGamal) [inline events_emitter/example-settings.py]: '*' Should be env.

File: events_emitter/example-settings.py [hunk 7]
```diff
@@ -161,5 +196,48 @@
 
 # Static files (CSS, JavaScript, Images)
 # https://docs.djangoproject.com/en/3.0/howto/static-files/
-
+STATIC_ROOT = 'staticfiles'
 STATIC_URL = '/static/'
+TABLE_NAME = getenv('TABLE_NAME')
+EVENTS_EMITTER_QUEUE = getenv('EVENTS_EMITTER_QUEUE', 'events_emitter')
+TIME_SERIES_TYPE = getenv('TIME_SERIES_TYPE')
+
```
Comment (AlSayedGamal) [inline events_emitter/example-settings.py]: Please, add sentry.

File: events_emitter/example-settings.py [hunk 8]
```diff
@@ -41,10 +43,14 @@
     'django.contrib.sessions',
     'django.contrib.messages',
     'django.contrib.staticfiles',
-    'events_emitter'
+    'events_emitter',
+    'django_celery_results',
+    'django_celery_beat',
+    'eventful_django',
```
Comment (AlSayedGamal) [inline events_emitter/example-settings.py]: Please, add Prometheus to the service settings and service dependencies in production.

File: events_emitter/middleware.py [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+        return self.get_response(request)
+
+    def liveness(self):
+        return HttpResponse("OK")
+
+    def readiness(self):
+        try:
+            with connection.cursor() as cursor:
+                cursor.execute("SELECT 1;")
```
Comment (AlSayedGamal) [inline events_emitter/middleware.py]: I think connection to redis is also required.

File: events_emitter/tasks.py [hunk 10]
```diff
@@ -0,0 +1,84 @@
+from __future__ import absolute_import
```
Comment (AlSayedGamal) [inline events_emitter/tasks.py]: File generally needs more DEBUG level logs.

Comment (amr-alaa95) [reply events_emitter/tasks.py]: In every level I put log for every rule and for every expression, So I think it's more enough logs

**Pull request merged**